# Teste Amostral da Linha 010

Código feito por Igor Laltuf, ago/23

---

## Objetivo

Este notebook tem como objetivo realizar o teste amostral da linha 010 com dados da primeira quinzena de janeiro de 2023. 

## Etapas
1) Comparar dados do gabarito enviado pelos consórcios com os dados da solução proposta pela DTDI.

2) Verificar se os veículos não identificados na solução operaram nos dias e horários apontados no gabarito, com base em dados de GPS e do reprocessamento de conformidade das viagens proveniente do dataset `rj-smtr-dev.20230818_projeto_subsidio_sppo_010`.

## Conclusão

Ao considerar um intervalo de 10 minutos para mais ou para menos entre os horários de partida, foram encontradas 699 das 706 viagens apresentadas no gabarito (99% das viagens do gabarito).

Após verificar as sete viagens não identificadas na solução acima, concluiu-se que:

- Não foram encontrados dados das quatro viagens do veículo A72037 na linha 010, mas foram encontrados sinais de GPS em horários compatíveis com duas das quatro viagens, operando na linha 007.

- No caso do veículo A72085 na viagem do dia 15/01/2023, a viagem ocorreu, mas ficou um pouco acima do intervalo de +-10 minutos e por isto não foi identificada anteriormente.

- No caso do veículo A72085 na viagem do dia 04/01/2023, não foram encontrados sinais de GPS no momento da viagem.

- No caso do veículo A72151 no dia 03/01/2023, a viagem ocorreu mas ficou um pouco acima do intervalo de +-10 minutos e por isto não foi identificada anteriormente.

Considerando as duas viagens identificadas e que ficaram acima do intervalo de +- 10 minutos, foram identificadas 701 das 706 viagens apresentadas no recurso.

obs: os dados do gabarito e da solução utilizados no teste amostral são provenientes da planilha disponível em https://docs.google.com/spreadsheets/d/11jKNeWoXB4Uke4WWwWsjHo8I-ZRr8f3Y/edit#gid=1849603428.

### Carregar bibliotecas 

In [159]:
import basedosdados as bd
import pandas as pd
import numpy as np
import plotly.express as px

from datetime import timedelta, datetime

# Paths
import os
import sys
from pathlib import Path 

current_path = Path().resolve()

paths = dict()
paths["raw"] = current_path / 'data' / 'raw'
paths["treated"] = current_path / 'data' / 'treated'

for path in paths.values():
    if not os.path.exists(path):
        os.makedirs(path)

# Seta o projeto a ser cobrado
bd.config.billing_project_id = "rj-smtr-dev"

# Seta as configurações a serem usadas
bd.config.project_config_path = "C:/Users/igorl/.basedosdados/"
# bd.config.project_config_path = "D:\\basedosdados\\dev"

### 1. Importar dados do gabarito e da solução

In [160]:
solucao = pd.read_excel('data/raw/amostra_linha_010_q1_2023_tratado.xlsx', sheet_name=0)  
solucao

data  servico sentido id_veiculo    datetime_partida  \
0   2023-01-02       10       I     A72037 2023-01-02 06:07:57   
1   2023-01-02       10       V     A72037 2023-01-02 06:32:00   
2   2023-01-02       10       I     A72085 2023-01-02 06:35:11   
3   2023-01-02       10       I     A72037 2023-01-02 07:00:04   
4   2023-01-02       10       V     A72085 2023-01-02 07:05:29   
..         ...      ...     ...        ...                 ...   
729 2023-01-15       10       V     A72085 2023-01-15 19:18:23   
730 2023-01-15       10       I     A72085 2023-01-15 19:42:45   
731 2023-01-15       10       V     A72085 2023-01-15 20:08:57   
732 2023-01-15       10       I     A72085 2023-01-15 20:32:29   
733 2023-01-15       10       V     A72085 2023-01-15 21:01:50   

       datetime_chegada  extensao  
0   2023-01-02 06:19:58      5505  
1   2023-01-02 06:43:21      4902  
2   2023-01-02 06:47:51      5505  
3   2023-01-02 07:13:44      5505  
4   2023-01-02 07:21:50      4902  
..                  ...       ...  
729 2023-01-15 19:32:43      4902  
730 2023-01-15 19:57:06      5505  
731 2023-01-15 20:20:57      4902  
732 2023-01-15 20:50:30      5505  
733 2023-01-15 21:25:35      4902  

[734 rows x 7 columns]

In [161]:
gabarito_original = pd.read_excel('data/raw/amostra_linha_010_q1_2023_tratado.xlsx', sheet_name=1) 
gabarito_original

data  servico sentido id_veiculo    datetime_partida  \
0   2023-01-02       10       I     A72037 2023-01-02 06:07:00   
1   2023-01-02       10       V     A72037 2023-01-02 06:29:00   
2   2023-01-02       10       I     A72085 2023-01-02 06:33:00   
3   2023-01-02       10       I     A72037 2023-01-02 06:58:00   
4   2023-01-02       10       V     A72085 2023-01-02 07:03:00   
..         ...      ...     ...        ...                 ...   
702 2023-01-15       10       I     A72085 2023-01-15 18:51:00   
703 2023-01-15       10       V     A72085 2023-01-15 19:16:00   
704 2023-01-15       10       I     A72085 2023-01-15 19:41:00   
705 2023-01-15       10       V     A72085 2023-01-15 20:06:00   
706 2023-01-15       10       I     A72085 2023-01-15 20:30:00   

       datetime_chegada  extensao  
0   2023-01-02 06:22:00      5505  
1   2023-01-02 06:56:00      4902  
2   2023-01-02 06:52:00      5505  
3   2023-01-02 07:16:00      5505  
4   2023-01-02 07:29:00      4902  
..                  ...       ...  
702 2023-01-15 19:14:00      5505  
703 2023-01-15 19:38:00      4902  
704 2023-01-15 19:59:00      5505  
705 2023-01-15 20:28:00      4902  
706 2023-01-15 20:53:00      5505  

[707 rows x 7 columns]

### 2. Tratar dados do gabarito

Remover os casos em que o mesmo veículo fez duas viagens no mesmo dia e horário.

In [162]:

gabarito = gabarito_original.sort_values(by=['id_veiculo', 'datetime_partida', 'datetime_chegada'])

# Filtrar linhas com base nas condições de sobreposição
mask = (
    gabarito.groupby('id_veiculo')['datetime_chegada'].shift(1).fillna(pd.Timestamp.min) <= gabarito['datetime_partida']
) & (
    gabarito['datetime_chegada'] <= gabarito.groupby('id_veiculo')['datetime_partida'].shift(-1).fillna(pd.Timestamp.max)
)
gabarito_tratado = gabarito[mask]

gabarito_tratado


data  servico sentido id_veiculo    datetime_partida  \
0   2023-01-02       10       I     A72037 2023-01-02 06:07:00   
1   2023-01-02       10       V     A72037 2023-01-02 06:29:00   
3   2023-01-02       10       I     A72037 2023-01-02 06:58:00   
5   2023-01-02       10       V     A72037 2023-01-02 07:28:00   
8   2023-01-02       10       I     A72037 2023-01-02 08:02:00   
..         ...      ...     ...        ...                 ...   
100 2023-01-03       10       I     A72151 2023-01-03 17:50:00   
102 2023-01-03       10       V     A72151 2023-01-03 18:27:00   
104 2023-01-03       10       I     A72151 2023-01-03 19:19:00   
105 2023-01-03       10       V     A72151 2023-01-03 20:18:00   
106 2023-01-03       10       I     A72151 2023-01-03 21:03:00   

       datetime_chegada  extensao  
0   2023-01-02 06:22:00      5505  
1   2023-01-02 06:56:00      4902  
3   2023-01-02 07:16:00      5505  
5   2023-01-02 07:58:00      4902  
8   2023-01-02 08:24:00      5505  
..                  ...       ...  
100 2023-01-03 18:25:00      5505  
102 2023-01-03 19:13:00      4902  
104 2023-01-03 20:17:00      5505  
105 2023-01-03 21:00:00      4902  
106 2023-01-03 21:20:00      5505  

[705 rows x 7 columns]

In [163]:
# Verificar viagens do mesmo veículo no mesmo dia e horário
sobrepostos = gabarito.merge(gabarito_tratado, how='left', indicator=True)
sobrepostos = sobrepostos[sobrepostos['_merge'] == 'left_only']

sobrepostos


data  servico sentido id_veiculo    datetime_partida  \
393 2023-01-03       10       V     A72085 2023-01-03 21:18:00   
394 2023-01-03       10       I     A72085 2023-01-03 21:20:00   

       datetime_chegada  extensao     _merge  
393 2023-01-03 21:37:00      4902  left_only  
394 2023-01-03 21:36:00      5505  left_only

In [164]:
# Modificar gabarito_tratado para incluir uma das linhas do df com viagens sobrepostas
gabarito_tratado = pd.concat([gabarito_tratado, sobrepostos.iloc[[0]]])
gabarito_tratado = gabarito_tratado[['id_veiculo', 'datetime_partida', 'datetime_chegada']]
gabarito_tratado


id_veiculo    datetime_partida    datetime_chegada
0       A72037 2023-01-02 06:07:00 2023-01-02 06:22:00
1       A72037 2023-01-02 06:29:00 2023-01-02 06:56:00
3       A72037 2023-01-02 06:58:00 2023-01-02 07:16:00
5       A72037 2023-01-02 07:28:00 2023-01-02 07:58:00
8       A72037 2023-01-02 08:02:00 2023-01-02 08:24:00
..         ...                 ...                 ...
102     A72151 2023-01-03 18:27:00 2023-01-03 19:13:00
104     A72151 2023-01-03 19:19:00 2023-01-03 20:17:00
105     A72151 2023-01-03 20:18:00 2023-01-03 21:00:00
106     A72151 2023-01-03 21:03:00 2023-01-03 21:20:00
393     A72085 2023-01-03 21:18:00 2023-01-03 21:37:00

[706 rows x 3 columns]

Na solução encontramos 734 viagens realizadas contra 706 viagens realizadas no gabarito após o tratamento.

### 3. Comparar as viagens do gabarito com as viagens da solução

Para realizar a comparação entre os dados, foi considerado um desvio de 10 minutos para mais ou para menos no horário de partida.

In [165]:
# Calcular intervalos de 10 minutos
gabarito_tratado['inicio_intervalo'] = gabarito_tratado['datetime_partida'] - pd.Timedelta(seconds=600)
gabarito_tratado['fim_intervalo'] = gabarito_tratado['datetime_partida'] + pd.Timedelta(seconds=600)

In [166]:
# 1. Faça um left join usando 'id_veiculo' como chave
merged_df = gabarito_tratado.merge(solucao, on='id_veiculo', how='left', suffixes=('_gabarito', '_solucao'))

# 2. Filtrar as linhas com base nas condições de intervalo de tempo
resultado = merged_df[
    (merged_df['datetime_partida_solucao'] >= merged_df['inicio_intervalo']) & 
    (merged_df['datetime_partida_solucao'] <= merged_df['fim_intervalo'])
]

In [167]:
# 3. Remover linhas duplicadas
resultado = resultado.drop_duplicates()

# 4. Selecionar colunas específicas
resultado = resultado[['data', 'servico', 'sentido', 'id_veiculo', 'datetime_partida_gabarito', 
                       'datetime_chegada_gabarito', 'extensao', 'inicio_intervalo', 'fim_intervalo', 
                       'datetime_partida_solucao', 'datetime_chegada_solucao']]

resultado


data  servico sentido id_veiculo datetime_partida_gabarito  \
0      2023-01-02       10       I     A72037       2023-01-02 06:07:00   
259    2023-01-02       10       V     A72037       2023-01-02 06:29:00   
518    2023-01-02       10       I     A72037       2023-01-02 06:58:00   
777    2023-01-02       10       V     A72037       2023-01-02 07:28:00   
1036   2023-01-02       10       I     A72037       2023-01-02 08:02:00   
...           ...      ...     ...        ...                       ...   
208428 2023-01-03       10       I     A72151       2023-01-03 17:50:00   
208436 2023-01-03       10       V     A72151       2023-01-03 18:27:00   
208452 2023-01-03       10       V     A72151       2023-01-03 20:18:00   
208460 2023-01-03       10       I     A72151       2023-01-03 21:03:00   
208517 2023-01-03       10       I     A72085       2023-01-03 21:18:00   

       datetime_chegada_gabarito  extensao    inicio_intervalo  \
0            2023-01-02 06:22:00      5505 2023-01-02 05:57:00   
259          2023-01-02 06:56:00      4902 2023-01-02 06:19:00   
518          2023-01-02 07:16:00      5505 2023-01-02 06:48:00   
777          2023-01-02 07:58:00      4902 2023-01-02 07:18:00   
1036         2023-01-02 08:24:00      5505 2023-01-02 07:52:00   
...                          ...       ...                 ...   
208428       2023-01-03 18:25:00      5505 2023-01-03 17:40:00   
208436       2023-01-03 19:13:00      4902 2023-01-03 18:17:00   
208452       2023-01-03 21:00:00      4902 2023-01-03 20:08:00   
208460       2023-01-03 21:20:00      5505 2023-01-03 20:53:00   
208517       2023-01-03 21:37:00      5505 2023-01-03 21:08:00   

             fim_intervalo datetime_partida_solucao datetime_chegada_solucao  
0      2023-01-02 06:17:00      2023-01-02 06:07:57      2023-01-02 06:19:58  
259    2023-01-02 06:39:00      2023-01-02 06:32:00      2023-01-02 06:43:21  
518    2023-01-02 07:08:00      2023-01-02 07:00:04      2023-01-02 07:13:44  
777    2023-01-02 07:38:00      2023-01-02 07:31:27      2023-01-02 07:47:47  
1036   2023-01-02 08:12:00      2023-01-02 08:03:43      2023-01-02 08:22:04  
...                    ...                      ...                      ...  
208428 2023-01-03 18:00:00      2023-01-03 17:51:34      2023-01-03 18:22:35  
208436 2023-01-03 18:37:00      2023-01-03 18:30:36      2023-01-03 18:59:52  
208452 2023-01-03 20:28:00      2023-01-03 20:21:27      2023-01-03 20:45:28  
208460 2023-01-03 21:13:00      2023-01-03 21:04:37      2023-01-03 21:18:38  
208517 2023-01-03 21:28:00      2023-01-03 21:21:36      2023-01-03 21:35:16  

[699 rows x 11 columns]

Foram encontradas 699 das 706 viagens apresentadas no gabarito (99% das viagens do gabarito).

### 4. Identificar as viagens que não foram encontradas na solução e que estão no gabarito

In [168]:
tabela = gabarito_tratado.merge(resultado, how = 'left', 
                                left_on=['id_veiculo', 'datetime_partida'], 
                                right_on=['id_veiculo', 'datetime_partida_gabarito'])

linhas_nan = tabela[tabela['sentido'].isna()]
linhas_nan

id_veiculo    datetime_partida    datetime_chegada  inicio_intervalo_x  \
51      A72037 2023-01-04 06:00:00 2023-01-04 06:16:00 2023-01-04 05:50:00   
52      A72037 2023-01-04 06:28:00 2023-01-04 07:01:00 2023-01-04 06:18:00   
53      A72037 2023-01-04 07:03:00 2023-01-04 07:26:00 2023-01-04 06:53:00   
54      A72037 2023-01-04 07:29:00 2023-01-04 07:53:00 2023-01-04 07:19:00   
394     A72085 2023-01-04 06:30:00 2023-01-04 06:50:00 2023-01-04 06:20:00   
672     A72085 2023-01-15 07:20:00 2023-01-15 07:55:00 2023-01-15 07:10:00   
702     A72151 2023-01-03 19:19:00 2023-01-03 20:17:00 2023-01-03 19:09:00   

        fim_intervalo_x data  servico sentido datetime_partida_gabarito  \
51  2023-01-04 06:10:00  NaT      NaN     NaN                       NaT   
52  2023-01-04 06:38:00  NaT      NaN     NaN                       NaT   
53  2023-01-04 07:13:00  NaT      NaN     NaN                       NaT   
54  2023-01-04 07:39:00  NaT      NaN     NaN                       NaT   
394 2023-01-04 06:40:00  NaT      NaN     NaN                       NaT   
672 2023-01-15 07:30:00  NaT      NaN     NaN                       NaT   
702 2023-01-03 19:29:00  NaT      NaN     NaN                       NaT   

    datetime_chegada_gabarito  extensao inicio_intervalo_y fim_intervalo_y  \
51                        NaT       NaN                NaT             NaT   
52                        NaT       NaN                NaT             NaT   
53                        NaT       NaN                NaT             NaT   
54                        NaT       NaN                NaT             NaT   
394                       NaT       NaN                NaT             NaT   
672                       NaT       NaN                NaT             NaT   
702                       NaT       NaN                NaT             NaT   

    datetime_partida_solucao datetime_chegada_solucao  
51                       NaT                      NaT  
52                       NaT                      NaT  
53                       NaT                      NaT  
54                       NaT                      NaT  
394                      NaT                      NaT  
672                      NaT                      NaT  
702                      NaT                      NaT

Sete viagens que constam no gabarito não foram encontradas na solução e foram realizadas por três veículos diferentes. 

1) Quatro viagens do veículo A72037 no início do dia 04/01/2023

In [169]:
tabela['datetime_partida'] = pd.to_datetime(tabela['datetime_partida'])

tabela_A72037 = tabela[(tabela['id_veiculo'] == 'A72037') & (tabela['datetime_partida'].dt.date == pd.Timestamp('2023-01-04').date())]

tabela_A72037 = tabela_A72037[['id_veiculo', 'datetime_partida','datetime_chegada','datetime_partida_solucao','datetime_chegada_solucao']]
tabela_A72037

id_veiculo    datetime_partida    datetime_chegada  \
51     A72037 2023-01-04 06:00:00 2023-01-04 06:16:00   
52     A72037 2023-01-04 06:28:00 2023-01-04 07:01:00   
53     A72037 2023-01-04 07:03:00 2023-01-04 07:26:00   
54     A72037 2023-01-04 07:29:00 2023-01-04 07:53:00   
55     A72037 2023-01-04 08:01:00 2023-01-04 08:24:00   
56     A72037 2023-01-04 08:33:00 2023-01-04 09:03:00   
57     A72037 2023-01-04 09:07:00 2023-01-04 09:30:00   
58     A72037 2023-01-04 09:32:00 2023-01-04 10:01:00   
59     A72037 2023-01-04 10:05:00 2023-01-04 10:33:00   
60     A72037 2023-01-04 10:34:00 2023-01-04 11:04:00   
61     A72037 2023-01-04 11:08:00 2023-01-04 11:34:00   
62     A72037 2023-01-04 11:37:00 2023-01-04 11:58:00   
63     A72037 2023-01-04 12:02:00 2023-01-04 12:28:00   
64     A72037 2023-01-04 12:32:00 2023-01-04 12:53:00   
65     A72037 2023-01-04 12:55:00 2023-01-04 13:26:00   
66     A72037 2023-01-04 13:31:00 2023-01-04 14:07:00   
67     A72037 2023-01-04 14:09:00 2023-01-04 14:38:00   
68     A72037 2023-01-04 14:41:00 2023-01-04 15:12:00   
69     A72037 2023-01-04 15:14:00 2023-01-04 15:51:00   
70     A72037 2023-01-04 15:52:00 2023-01-04 16:25:00   
71     A72037 2023-01-04 16:27:00 2023-01-04 16:59:00   
72     A72037 2023-01-04 17:01:00 2023-01-04 17:36:00   
73     A72037 2023-01-04 17:39:00 2023-01-04 18:11:00   
74     A72037 2023-01-04 18:13:00 2023-01-04 18:49:00   
75     A72037 2023-01-04 18:49:00 2023-01-04 19:15:00   
76     A72037 2023-01-04 19:18:00 2023-01-04 19:45:00   
77     A72037 2023-01-04 19:47:00 2023-01-04 20:07:00   
78     A72037 2023-01-04 20:14:00 2023-01-04 20:41:00   
79     A72037 2023-01-04 20:42:00 2023-01-04 21:00:00   

   datetime_partida_solucao datetime_chegada_solucao  
51                      NaT                      NaT  
52                      NaT                      NaT  
53                      NaT                      NaT  
54                      NaT                      NaT  
55      2023-01-04 08:03:10      2023-01-04 08:22:11  
56      2023-01-04 08:38:58      2023-01-04 08:57:58  
57      2023-01-04 09:09:59      2023-01-04 09:28:19  
58      2023-01-04 09:38:20      2023-01-04 09:56:40  
59      2023-01-04 10:08:41      2023-01-04 10:28:42  
60      2023-01-04 10:40:02      2023-01-04 10:59:22  
61      2023-01-04 11:10:23      2023-01-04 11:32:24  
62      2023-01-04 11:41:24      2023-01-04 11:54:44  
63      2023-01-04 12:04:45      2023-01-04 12:26:25  
64      2023-01-04 12:35:33      2023-01-04 12:48:33  
65      2023-01-04 12:58:34      2023-01-04 13:22:54  
66      2023-01-04 13:34:09      2023-01-04 13:51:30  
67      2023-01-04 14:14:24      2023-01-04 14:36:44  
68      2023-01-04 14:44:25      2023-01-04 15:04:05  
69      2023-01-04 15:16:54      2023-01-04 15:46:35  
70      2023-01-04 15:56:36      2023-01-04 16:18:16  
71      2023-01-04 16:28:33      2023-01-04 16:56:53  
72      2023-01-04 17:04:54      2023-01-04 17:22:54  
73      2023-01-04 17:41:47      2023-01-04 18:07:08  
74      2023-01-04 18:16:28      2023-01-04 18:38:49  
75      2023-01-04 18:51:34      2023-01-04 19:11:35  
76      2023-01-04 19:20:15      2023-01-04 19:35:16  
77      2023-01-04 19:48:45      2023-01-04 20:05:06  
78      2023-01-04 20:17:58      2023-01-04 20:28:38  
79      2023-01-04 20:44:08      2023-01-04 20:58:48

Ao verificar a tabela de conformidade reprocessada para janeiro, também não foram encontradas as viagens antes de 08:00.

In [170]:
q = f"""
    SELECT
       *
     FROM
       `rj-smtr-dev.20230818_projeto_subsidio_sppo_010.viagem_conformidade`
      WHERE
       data = "2023-01-04"
       AND id_veiculo = 'A72037'
"""

registros_conformidade_A72037 = bd.read_sql(q)

Downloading: 100%|██████████| 26/26 [00:00<00:00, 76.03rows/s]


In [171]:
registros_conformidade_A72037 = registros_conformidade_A72037[['id_veiculo','datetime_partida','datetime_chegada','servico_informado']]
registros_conformidade_A72037.sort_values(by='datetime_partida')


id_veiculo    datetime_partida    datetime_chegada servico_informado
12     A72037 2023-01-04 08:03:10 2023-01-04 08:22:11               010
11     A72037 2023-01-04 08:38:58 2023-01-04 08:57:58               010
10     A72037 2023-01-04 09:09:59 2023-01-04 09:28:19               010
9      A72037 2023-01-04 09:38:20 2023-01-04 09:56:40               010
16     A72037 2023-01-04 10:08:41 2023-01-04 10:28:42               010
13     A72037 2023-01-04 10:40:02 2023-01-04 10:59:22               010
19     A72037 2023-01-04 11:10:23 2023-01-04 11:32:24               010
3      A72037 2023-01-04 11:41:24 2023-01-04 11:54:44               010
18     A72037 2023-01-04 12:04:45 2023-01-04 12:26:25               010
2      A72037 2023-01-04 12:35:33 2023-01-04 12:48:33               010
22     A72037 2023-01-04 12:58:34 2023-01-04 13:22:54               010
7      A72037 2023-01-04 13:34:09 2023-01-04 13:51:30               010
21     A72037 2023-01-04 14:14:24 2023-01-04 14:36:44               010
14     A72037 2023-01-04 14:44:25 2023-01-04 15:04:05               010
25     A72037 2023-01-04 15:16:54 2023-01-04 15:46:35               010
17     A72037 2023-01-04 15:56:36 2023-01-04 16:18:16               010
24     A72037 2023-01-04 16:28:33 2023-01-04 16:56:53               010
8      A72037 2023-01-04 17:04:54 2023-01-04 17:22:54               010
23     A72037 2023-01-04 17:41:47 2023-01-04 18:07:08               010
20     A72037 2023-01-04 18:16:28 2023-01-04 18:38:49               010
15     A72037 2023-01-04 18:51:34 2023-01-04 19:11:35               010
5      A72037 2023-01-04 19:20:15 2023-01-04 19:35:16               010
6      A72037 2023-01-04 19:48:45 2023-01-04 20:05:06               010
1      A72037 2023-01-04 20:17:58 2023-01-04 20:28:38               010
4      A72037 2023-01-04 20:44:08 2023-01-04 20:58:48               010
0      A72037 2023-01-04 21:08:26 2023-01-04 21:17:27               010

In [172]:
q = f"""
SELECT
  *
FROM
  `rj-smtr.br_rj_riodejaneiro_veiculos.gps_sppo`
WHERE
  DATA = "2023-01-04"
  AND id_veiculo = 'A72037'
  AND flag_em_operacao = true
"""

registros_gps_A72037 = bd.read_sql(q)

Downloading: 100%|██████████| 2577/2577 [00:00<00:00, 2791.98rows/s]


In [173]:
registros_gps_A72037 = registros_gps_A72037[['id_veiculo', 'hora','servico']]
registros_gps_A72037.sort_values(by='hora')

id_veiculo      hora servico
926      A72037  06:47:25     007
2112     A72037  06:47:45     007
1375     A72037  06:48:06     007
1965     A72037  06:48:26     007
1423     A72037  06:48:46     007
...         ...       ...     ...
311      A72037  21:38:04     010
819      A72037  21:38:24     010
1057     A72037  21:38:44     010
1361     A72037  21:39:04     010
2485     A72037  21:39:24     010

[2577 rows x 3 columns]

Ao verificar a tabela com os sinais de GPS, entre 06:47 e 07:45, intervalo que é compatível com duas das quatro viagens não identificadas, foram identificados sinais do veículo operando no serviço 007, e não no serviço 010. Ou seja, das quatro viagens, duas não foram identificadas e as outras duas foram realizadas em outro serviço.

2) Duas viagens do veículo A72085: uma no dia 04/01/2023 e outra no dia 15/01/2023

2.1) Viagem do veículo A72085 no dia 04/01/2023

In [174]:
tabela_A72085_04 = tabela[(tabela['id_veiculo'] == 'A72085') & (tabela['datetime_partida'].dt.date == pd.Timestamp('2023-01-04').date())]

tabela_A72085_04 = tabela_A72085_04[['id_veiculo', 'datetime_partida','datetime_chegada','datetime_partida_solucao','datetime_chegada_solucao']]
tabela_A72085_04

id_veiculo    datetime_partida    datetime_chegada  \
394     A72085 2023-01-04 06:30:00 2023-01-04 06:50:00   
395     A72085 2023-01-04 06:59:00 2023-01-04 07:28:00   
396     A72085 2023-01-04 07:30:00 2023-01-04 07:50:00   
397     A72085 2023-01-04 07:59:00 2023-01-04 08:29:00   
398     A72085 2023-01-04 08:32:00 2023-01-04 09:00:00   
399     A72085 2023-01-04 09:03:00 2023-01-04 09:31:00   
400     A72085 2023-01-04 09:34:00 2023-01-04 10:04:00   
401     A72085 2023-01-04 10:07:00 2023-01-04 10:33:00   
402     A72085 2023-01-04 10:35:00 2023-01-04 11:04:00   
403     A72085 2023-01-04 11:06:00 2023-01-04 11:27:00   
404     A72085 2023-01-04 11:30:00 2023-01-04 11:58:00   
405     A72085 2023-01-04 12:03:00 2023-01-04 12:30:00   
406     A72085 2023-01-04 12:32:00 2023-01-04 13:03:00   
407     A72085 2023-01-04 13:05:00 2023-01-04 13:37:00   
408     A72085 2023-01-04 13:39:00 2023-01-04 14:05:00   
409     A72085 2023-01-04 14:06:00 2023-01-04 14:34:00   
410     A72085 2023-01-04 14:35:00 2023-01-04 15:04:00   
411     A72085 2023-01-04 15:06:00 2023-01-04 15:34:00   
412     A72085 2023-01-04 15:36:00 2023-01-04 16:08:00   
413     A72085 2023-01-04 16:10:00 2023-01-04 16:38:00   
414     A72085 2023-01-04 16:40:00 2023-01-04 17:07:00   
415     A72085 2023-01-04 17:16:00 2023-01-04 17:49:00   
416     A72085 2023-01-04 17:51:00 2023-01-04 18:15:00   
417     A72085 2023-01-04 18:22:00 2023-01-04 19:04:00   
418     A72085 2023-01-04 19:06:00 2023-01-04 19:28:00   
419     A72085 2023-01-04 19:38:00 2023-01-04 20:02:00   
420     A72085 2023-01-04 20:09:00 2023-01-04 20:29:00   
421     A72085 2023-01-04 20:32:00 2023-01-04 20:54:00   
422     A72085 2023-01-04 20:56:00 2023-01-04 21:14:00   

    datetime_partida_solucao datetime_chegada_solucao  
394                      NaT                      NaT  
395      2023-01-04 07:01:53      2023-01-04 07:14:14  
396      2023-01-04 07:31:57      2023-01-04 07:48:17  
397      2023-01-04 08:03:39      2023-01-04 08:22:40  
398      2023-01-04 08:34:00      2023-01-04 08:56:41  
399      2023-01-04 09:06:01      2023-01-04 09:22:02  
400      2023-01-04 09:37:39      2023-01-04 10:02:20  
401      2023-01-04 10:10:20      2023-01-04 10:28:01  
402      2023-01-04 10:38:01      2023-01-04 11:01:02  
403      2023-01-04 11:10:42      2023-01-04 11:25:03  
404      2023-01-04 11:32:03      2023-01-04 11:56:04  
405      2023-01-04 12:06:04      2023-01-04 12:25:24  
406      2023-01-04 12:34:45      2023-01-04 13:00:25  
407      2023-01-04 13:09:46      2023-01-04 13:28:06  
408      2023-01-04 13:42:47      2023-01-04 14:02:27  
409      2023-01-04 14:10:27      2023-01-04 14:27:08  
410      2023-01-04 14:38:29      2023-01-04 15:00:49  
411      2023-01-04 15:09:10      2023-01-04 15:25:30  
412      2023-01-04 15:40:21      2023-01-04 16:05:22  
413      2023-01-04 16:13:42      2023-01-04 16:32:23  
414      2023-01-04 16:43:39      2023-01-04 17:05:00  
415      2023-01-04 17:18:48      2023-01-04 17:35:28  
416      2023-01-04 17:57:09      2023-01-04 18:13:09  
417      2023-01-04 18:24:38      2023-01-04 18:39:39  
418      2023-01-04 19:07:38      2023-01-04 19:24:59  
419      2023-01-04 19:40:27      2023-01-04 19:53:07  
420      2023-01-04 20:11:04      2023-01-04 20:27:05  
421      2023-01-04 20:35:36      2023-01-04 20:47:17  
422      2023-01-04 20:57:35      2023-01-04 21:13:16

A viagem não foi encontrada na tabela de conformidade reprocessada para janeiro.

In [175]:
q = f"""
    SELECT
       *
     FROM
       `rj-smtr-dev.20230818_projeto_subsidio_sppo_010.viagem_conformidade`
      WHERE
       data = "2023-01-04"
       AND id_veiculo = 'A72085'
"""

registros_conformidade_A72085_04 = bd.read_sql(q)

Downloading: 100%|██████████| 29/29 [00:00<00:00, 81.69rows/s]


In [176]:
registros_conformidade_A72085_04 = registros_conformidade_A72085_04[['id_veiculo', 'datetime_partida','datetime_chegada']]
registros_conformidade_A72085_04.sort_values(by='datetime_partida')

id_veiculo    datetime_partida    datetime_chegada
2      A72085 2023-01-04 07:01:53 2023-01-04 07:14:14
11     A72085 2023-01-04 07:31:57 2023-01-04 07:48:17
18     A72085 2023-01-04 08:03:39 2023-01-04 08:22:40
23     A72085 2023-01-04 08:34:00 2023-01-04 08:56:41
7      A72085 2023-01-04 09:06:01 2023-01-04 09:22:02
26     A72085 2023-01-04 09:37:39 2023-01-04 10:02:20
15     A72085 2023-01-04 10:10:20 2023-01-04 10:28:01
24     A72085 2023-01-04 10:38:01 2023-01-04 11:01:02
4      A72085 2023-01-04 11:10:42 2023-01-04 11:25:03
25     A72085 2023-01-04 11:32:03 2023-01-04 11:56:04
19     A72085 2023-01-04 12:06:04 2023-01-04 12:25:24
28     A72085 2023-01-04 12:34:45 2023-01-04 13:00:25
16     A72085 2023-01-04 13:09:46 2023-01-04 13:28:06
20     A72085 2023-01-04 13:42:47 2023-01-04 14:02:27
13     A72085 2023-01-04 14:10:27 2023-01-04 14:27:08
22     A72085 2023-01-04 14:38:29 2023-01-04 15:00:49
10     A72085 2023-01-04 15:09:10 2023-01-04 15:25:30
27     A72085 2023-01-04 15:40:21 2023-01-04 16:05:22
17     A72085 2023-01-04 16:13:42 2023-01-04 16:32:23
21     A72085 2023-01-04 16:43:39 2023-01-04 17:05:00
12     A72085 2023-01-04 17:18:48 2023-01-04 17:35:28
9      A72085 2023-01-04 17:57:09 2023-01-04 18:13:09
5      A72085 2023-01-04 18:24:38 2023-01-04 18:39:39
14     A72085 2023-01-04 19:07:38 2023-01-04 19:24:59
3      A72085 2023-01-04 19:40:27 2023-01-04 19:53:07
8      A72085 2023-01-04 20:11:04 2023-01-04 20:27:05
1      A72085 2023-01-04 20:35:36 2023-01-04 20:47:17
6      A72085 2023-01-04 20:57:35 2023-01-04 21:13:16
0      A72085 2023-01-04 21:21:55 2023-01-04 21:33:36

In [177]:
q = f"""
SELECT
  *
FROM
  `rj-smtr.br_rj_riodejaneiro_veiculos.gps_sppo`
WHERE
  DATA = "2023-01-04"
  AND id_veiculo = 'A72085'
  AND flag_em_operacao = true
"""

registros_gps_A72085_04 = bd.read_sql(q)

Downloading: 100%|██████████| 2538/2538 [00:00<00:00, 2887.41rows/s]


In [178]:
registros_gps_A72085_04 = registros_gps_A72085_04[['id_veiculo', 'hora','servico']]
registros_gps_A72085_04.sort_values(by='hora')

id_veiculo      hora servico
566      A72085  06:47:41     010
1456     A72085  06:48:01     010
1429     A72085  06:48:21     010
122      A72085  06:48:41     010
1612     A72085  06:49:01     010
...         ...       ...     ...
1751     A72085  21:59:27     010
1925     A72085  21:59:47     010
1805     A72085  22:00:07     010
870      A72085  22:00:27     010
2460     A72085  22:00:47     010

[2538 rows x 3 columns]

Foi identificado sinal de GPS em operação para o veículo A72085 desde 06:47, horário próximo ao horário do final da viagem não identificada (06:50). Os sinais de GPS não contemplaram a duração total da viagem.

2.2) Viagem do veículo A72085 no dia 15/01/2023

In [179]:
tabela_A72085_15 = tabela[(tabela['id_veiculo'] == 'A72085') & (tabela['datetime_partida'].dt.date == pd.Timestamp('2023-01-15').date())]

tabela_A72085_15 = tabela_A72085_15[['id_veiculo', 'datetime_partida','datetime_chegada','datetime_partida_solucao','datetime_chegada_solucao']]
tabela_A72085_15
tabela_A72085_15.shape[0]


31

In [180]:
q = f"""
    SELECT
       *
     FROM
       `rj-smtr-dev.20230818_projeto_subsidio_sppo_010.viagem_conformidade`
      WHERE
       data = "2023-01-15"
       AND id_veiculo = 'A72085'
"""

registros_conformidade_A72085_15 = bd.read_sql(q)
registros_conformidade_A72085_15.shape[0]

Downloading: 100%|██████████| 32/32 [00:00<00:00, 91.43rows/s]


32

In [181]:
registros_conformidade_A72085_15 = registros_conformidade_A72085_15[['id_veiculo', 'datetime_partida','datetime_chegada']]
registros_conformidade_A72085_15.sort_values(by='datetime_partida')

id_veiculo    datetime_partida    datetime_chegada
16     A72085 2023-01-15 06:37:54 2023-01-15 06:50:34
1      A72085 2023-01-15 06:54:54 2023-01-15 07:03:34
28     A72085 2023-01-15 07:06:55 2023-01-15 07:17:15
12     A72085 2023-01-15 07:35:22 2023-01-15 07:47:02
27     A72085 2023-01-15 08:03:32 2023-01-15 08:15:12
3      A72085 2023-01-15 08:31:13 2023-01-15 08:43:53
26     A72085 2023-01-15 09:08:37 2023-01-15 09:20:17
9      A72085 2023-01-15 09:35:39 2023-01-15 09:46:39
24     A72085 2023-01-15 10:04:37 2023-01-15 10:16:37
15     A72085 2023-01-15 10:30:31 2023-01-15 10:44:51
18     A72085 2023-01-15 11:00:27 2023-01-15 11:10:27
13     A72085 2023-01-15 11:26:51 2023-01-15 11:41:32
21     A72085 2023-01-15 11:55:16 2023-01-15 12:11:37
4      A72085 2023-01-15 12:32:07 2023-01-15 12:43:27
31     A72085 2023-01-15 12:54:08 2023-01-15 13:07:48
0      A72085 2023-01-15 13:27:52 2023-01-15 13:43:32
23     A72085 2023-01-15 13:58:36 2023-01-15 14:13:17
7      A72085 2023-01-15 14:25:57 2023-01-15 14:40:58
17     A72085 2023-01-15 14:50:59 2023-01-15 15:06:59
6      A72085 2023-01-15 15:18:20 2023-01-15 15:33:00
19     A72085 2023-01-15 15:43:41 2023-01-15 16:01:32
11     A72085 2023-01-15 16:14:13 2023-01-15 16:31:33
25     A72085 2023-01-15 16:42:14 2023-01-15 16:57:34
5      A72085 2023-01-15 17:05:54 2023-01-15 17:33:20
22     A72085 2023-01-15 17:46:33 2023-01-15 18:06:13
14     A72085 2023-01-15 18:13:54 2023-01-15 18:35:34
30     A72085 2023-01-15 18:52:02 2023-01-15 19:11:03
2      A72085 2023-01-15 19:18:23 2023-01-15 19:32:43
29     A72085 2023-01-15 19:42:45 2023-01-15 19:57:06
10     A72085 2023-01-15 20:08:57 2023-01-15 20:20:57
20     A72085 2023-01-15 20:32:29 2023-01-15 20:50:30
8      A72085 2023-01-15 21:01:50 2023-01-15 21:25:35

No dia 15/01 para o mesmo veículo, foi encontrada a mesma quantidade de viagens entre a tabela de conformidade e a tabela com a solução. A viagem não foi identificada anteriormente, pois o horário de partida dela é maior do que o intervalo de +- 10 minutos. A viagem de fato ocorreu.

3) Uma viagem do veículo A72151 no dia 03/01/2023

In [182]:
tabela_A72151 = tabela[(tabela['id_veiculo'] == 'A72151') & (tabela['datetime_partida'].dt.date == pd.Timestamp('2023-01-03').date())]
tabela_A72151 = tabela_A72151[['id_veiculo', 'datetime_partida','datetime_chegada','datetime_partida_solucao','datetime_chegada_solucao']]
tabela_A72151

id_veiculo    datetime_partida    datetime_chegada  \
700     A72151 2023-01-03 17:50:00 2023-01-03 18:25:00   
701     A72151 2023-01-03 18:27:00 2023-01-03 19:13:00   
702     A72151 2023-01-03 19:19:00 2023-01-03 20:17:00   
703     A72151 2023-01-03 20:18:00 2023-01-03 21:00:00   
704     A72151 2023-01-03 21:03:00 2023-01-03 21:20:00   

    datetime_partida_solucao datetime_chegada_solucao  
700      2023-01-03 17:51:34      2023-01-03 18:22:35  
701      2023-01-03 18:30:36      2023-01-03 18:59:52  
702                      NaT                      NaT  
703      2023-01-03 20:21:27      2023-01-03 20:45:28  
704      2023-01-03 21:04:37      2023-01-03 21:18:38

Checar a tabela de conformidade

In [183]:
q = f"""
    SELECT
       *
     FROM
       `rj-smtr-dev.20230818_projeto_subsidio_sppo_010.viagem_conformidade`
      WHERE
       data = "2023-01-03"
       AND id_veiculo = 'A72151'
"""

registros_conformidade_A72151 = bd.read_sql(q)

Downloading: 100%|██████████| 21/21 [00:00<00:00, 64.02rows/s]


In [184]:
registros_conformidade_A72151 = registros_conformidade_A72151[registros_conformidade_A72151['servico_informado'] == '010']
registros_conformidade_A72151 = registros_conformidade_A72151.sort_values(by = 'datetime_partida')
registros_conformidade_A72151 = registros_conformidade_A72151[['id_veiculo', 'datetime_partida','datetime_chegada']]
registros_conformidade_A72151

id_veiculo    datetime_partida    datetime_chegada
17     A72151 2023-01-03 17:51:34 2023-01-03 18:22:35
15     A72151 2023-01-03 18:30:36 2023-01-03 18:59:52
20     A72151 2023-01-03 19:30:25 2023-01-03 20:15:42
4      A72151 2023-01-03 20:21:27 2023-01-03 20:45:28
0      A72151 2023-01-03 21:04:37 2023-01-03 21:18:38
1      A72151 2023-01-03 21:24:38 2023-01-03 21:43:39

A viagem ocorreu, mas o datetime_partida estava acima do intervalo de +-10 minutos.

5. Viagens que foram encontradas na solução, mas que não foram encontradas no gabarito.

Trinta e cinco viagens que constam na solução não foram encontradas no gabarito.

In [185]:
tabela = solucao.merge(resultado, how = 'left', 
                                left_on=['id_veiculo', 'datetime_partida'], 
                                right_on=['id_veiculo', 'datetime_partida_solucao'])

tabela

linhas_nan = tabela[tabela['sentido_y'].isna()]
linhas_nan
len(linhas_nan)

35

### 6. Conclusão

Após verificar as sete viagens não identificadas na solução acima, concluiu-se que:

- Não foram encontrados dados das quatro viagens do veículo A72037 na linha 010, mas foram encontrados sinais de GPS em horários compatíveis com duas das quatro viagens, operando na linha 007.

- No caso do veículo A72085 na viagem do dia 15/01/2023, a viagem ocorreu, mas ficou um pouco acima do intervalo de +-10 minutos e por isto não foi identificada anteriormente.

- No caso do veículo A72085 na viagem do dia 04/01/2023, não foram encontrados sinais de GPS no momento da viagem.

- No caso do veículo A72151 no dia 03/01/2023, a viagem ocorreu mas ficou um pouco acima do intervalo de +-10 minutos e por isto não foi identificada anteriormente.

### 7. Exportar planilha com o status das viagens do gabarito

In [186]:
# 1 - viagens identificadas

# selecionar a tabela com viagens identificadas

viagens_identificadas = resultado[['id_veiculo','datetime_partida_gabarito']]
viagens_identificadas = viagens_identificadas.rename(columns={'datetime_partida_gabarito': 'datetime_partida'})

# add a coluna status
viagens_identificadas['status'] = 'O veículo existe e operou na linha indicada pelo recurso'
viagens_identificadas

id_veiculo    datetime_partida  \
0          A72037 2023-01-02 06:07:00   
259        A72037 2023-01-02 06:29:00   
518        A72037 2023-01-02 06:58:00   
777        A72037 2023-01-02 07:28:00   
1036       A72037 2023-01-02 08:02:00   
...           ...                 ...   
208428     A72151 2023-01-03 17:50:00   
208436     A72151 2023-01-03 18:27:00   
208452     A72151 2023-01-03 20:18:00   
208460     A72151 2023-01-03 21:03:00   
208517     A72085 2023-01-03 21:18:00   

                                                   status  
0       O veículo existe e operou na linha indicada pe...  
259     O veículo existe e operou na linha indicada pe...  
518     O veículo existe e operou na linha indicada pe...  
777     O veículo existe e operou na linha indicada pe...  
1036    O veículo existe e operou na linha indicada pe...  
...                                                   ...  
208428  O veículo existe e operou na linha indicada pe...  
208436  O veículo existe e operou na linha indicada pe...  
208452  O veículo existe e operou na linha indicada pe...  
208460  O veículo existe e operou na linha indicada pe...  
208517  O veículo existe e operou na linha indicada pe...  

[699 rows x 3 columns]

In [187]:
# 2 - Viagens inválidas
# classificar aquela 1 viagem como "viagem inválida (mesmo veículo fez duas viagens simultâneas)"
viagem_invalida = sobrepostos.iloc[1:2]
viagem_invalida = viagem_invalida[['id_veiculo','datetime_partida']]
viagem_invalida['status'] = 'Viagem inválida - o mesmo veículo fez duas viagens simultâneas'
viagem_invalida

id_veiculo    datetime_partida  \
394     A72085 2023-01-03 21:20:00   

                                                status  
394  Viagem inválida - o mesmo veículo fez duas via...

In [193]:
# 3 - Viagens veículo do A72037
# não emitiu sinal de GPS no momento da viagem
viagem_nao_encontrada = tabela_A72037.iloc[0:2]
viagem_nao_encontrada = viagem_nao_encontrada[['id_veiculo','datetime_partida']]
viagem_nao_encontrada['status'] = 'Viagem não encontrada - veículo não emitiu sinal de GPS durante a viagem'
viagem_nao_encontrada

id_veiculo    datetime_partida  \
51     A72037 2023-01-04 06:00:00   
52     A72037 2023-01-04 06:28:00   

                                               status  
51  Viagem não encontrada - veículo não emitiu sin...  
52  Viagem não encontrada - veículo não emitiu sin...

In [192]:
# operou em outro serviço (007)
viagem_em_outro_servico = tabela_A72037.iloc[2:4]
viagem_em_outro_servico = viagem_em_outro_servico[['id_veiculo','datetime_partida']]
viagem_em_outro_servico['status'] = 'Veículo operou, mas a viagem não foi no serviço indicado pelo recurso'
viagem_em_outro_servico 

id_veiculo    datetime_partida  \
53     A72037 2023-01-04 07:03:00   
54     A72037 2023-01-04 07:29:00   

                                               status  
53  Veículo operou, mas a viagem não foi no serviç...  
54  Veículo operou, mas a viagem não foi no serviç...

In [194]:
# 4 - Viagens veículo do veículo A72085 nos dias 04 e 15/01

# dia 04: Ônibus não emitiu sinal de GPS no momento na viagem.
viagem_nao_encontrada2 = tabela_A72085_04.iloc[0:1]
viagem_nao_encontrada2 = viagem_nao_encontrada2[['id_veiculo','datetime_partida']]
viagem_nao_encontrada2['status'] = 'Viagem não encontrada - veículo não emitiu sinal de GPS durante a viagem'
viagem_nao_encontrada2 

id_veiculo    datetime_partida  \
394     A72085 2023-01-04 06:30:00   

                                                status  
394  Viagem não encontrada - veículo não emitiu sin...

In [195]:
# dia 15 a viagem ocorreu acima dos 10 minutos esperados
viagem_identificada2 = tabela_A72085_15.iloc[3:4]
viagem_identificada2 = viagem_identificada2[['id_veiculo','datetime_partida']]
viagem_identificada2['status'] = 'O veículo existe e operou na linha indicada pelo recurso'
viagem_identificada2

id_veiculo    datetime_partida  \
672     A72085 2023-01-15 07:20:00   

                                                status  
672  O veículo existe e operou na linha indicada pe...

In [196]:
# 5 - Viagem do veículo A72151 ocorreu acima dos 10 minutos esperados
viagem_identificada3 = tabela_A72151.iloc[2:3]
viagem_identificada3 = viagem_identificada3[['id_veiculo','datetime_partida']]
viagem_identificada3['status'] = 'O veículo existe e operou na linha indicada pelo recurso'
viagem_identificada3

id_veiculo    datetime_partida  \
702     A72151 2023-01-03 19:19:00   

                                                status  
702  O veículo existe e operou na linha indicada pe...

In [197]:
tabela_status = pd.concat([viagens_identificadas, viagem_identificada2, viagem_identificada3, 
                           viagem_nao_encontrada, viagem_nao_encontrada2, viagem_invalida,
                           viagem_em_outro_servico], ignore_index=True)


tabela_final = pd.merge(gabarito_original, tabela_status, on = ['id_veiculo','datetime_partida'], how = 'left')
tabela_final

data  servico sentido id_veiculo    datetime_partida  \
0   2023-01-02       10       I     A72037 2023-01-02 06:07:00   
1   2023-01-02       10       V     A72037 2023-01-02 06:29:00   
2   2023-01-02       10       I     A72085 2023-01-02 06:33:00   
3   2023-01-02       10       I     A72037 2023-01-02 06:58:00   
4   2023-01-02       10       V     A72085 2023-01-02 07:03:00   
..         ...      ...     ...        ...                 ...   
702 2023-01-15       10       I     A72085 2023-01-15 18:51:00   
703 2023-01-15       10       V     A72085 2023-01-15 19:16:00   
704 2023-01-15       10       I     A72085 2023-01-15 19:41:00   
705 2023-01-15       10       V     A72085 2023-01-15 20:06:00   
706 2023-01-15       10       I     A72085 2023-01-15 20:30:00   

       datetime_chegada  extensao  \
0   2023-01-02 06:22:00      5505   
1   2023-01-02 06:56:00      4902   
2   2023-01-02 06:52:00      5505   
3   2023-01-02 07:16:00      5505   
4   2023-01-02 07:29:00      4902   
..                  ...       ...   
702 2023-01-15 19:14:00      5505   
703 2023-01-15 19:38:00      4902   
704 2023-01-15 19:59:00      5505   
705 2023-01-15 20:28:00      4902   
706 2023-01-15 20:53:00      5505   

                                                status  
0    O veículo existe e operou na linha indicada pe...  
1    O veículo existe e operou na linha indicada pe...  
2    O veículo existe e operou na linha indicada pe...  
3    O veículo existe e operou na linha indicada pe...  
4    O veículo existe e operou na linha indicada pe...  
..                                                 ...  
702  O veículo existe e operou na linha indicada pe...  
703  O veículo existe e operou na linha indicada pe...  
704  O veículo existe e operou na linha indicada pe...  
705  O veículo existe e operou na linha indicada pe...  
706  O veículo existe e operou na linha indicada pe...  

[707 rows x 8 columns]

In [ ]:
tabela_final.to_excel('data/output/tabela_gabarito_status.xlsx', index=False, engine='openpyxl')